os/board/rtl8721csm/src/component/common/api/wifi/wifi_conf.c

```diff
@@ -1303,6 +1305,18 @@ int wifi_on(rtw_mode_t mode)
 	static int event_init = 0;
 
 	device_mutex_lock(RT_DEV_LOCK_WLAN);

```

Comment:
only the thread named wifi msg handler call wifi_on(). so If you consider access from various upper layer threads, you don't need to do.

Reply:
hi @pillip8282 , adding WiFiRegisterLinkCallback() here in wifi_on() is to solve the "No response to app in TP1x SoftAP deassoc" issue. Because wifi_off() will set g_link_up and g_link_down to NULL, so we need to add WiFiRegisterLinkCallback() as a counterpart in wifi_on(). Thank you~